### PR TITLE
lakectl use of pre-sign capabilities

### LINKED ISSUE #5067: CLI to support presign
The following commands that enable `direct` upload/download capability can use `pre-sign`.
Adding `--pre-sign` flag will enable using the pre-sign link to transfer the data from the client the underlying storage.

 - lakectl fs upload
 - lakectl fs download
 - lakectl fs cat
 - lakectl fs stat

### PR DESCRIPTION
**Pull request by nopcoder**
Close #5067

TBD adding lakectl tests to cover the use of `--pre-sign`

### CODE REVIEW

File: api/swagger.yml [hunk 1]
```diff
@@ -860,6 +860,7 @@ components:
         - blockstore_type
         - blockstore_namespace_example
         - blockstore_namespace_ValidityRegex
+        - pre_sign_support
```
Comment (itaiad200) [inline api/swagger.yml:863]: Does our config block `pre_sign`? I thought it's enabled by default.

File: pkg/api/helpers/upload.go [hunk 2]
```diff
@@ -3,31 +3,29 @@ package helpers
...
 	"github.com/treeverse/lakefs/pkg/api"
 )
 
 // ClientUpload uploads contents as a file using client-side ("direct") access to underlying
 // storage.  It requires credentials both to lakeFS and to underlying storage, but
 // considerably reduces the load on the lakeFS server.
-func ClientUpload(ctx context.Context, client api.ClientWithResponsesInterface, repoID, branchID, filePath string, metadata map[string]string, contentType string, contents io.ReadSeeker) (*api.ObjectStats, error) {
-	resp, err := client.GetPhysicalAddressWithResponse(ctx, repoID, branchID, &api.GetPhysicalAddressParams{
-		Path: filePath,
+func ClientUpload(ctx context.Context, client api.ClientWithResponsesInterface, repoID, branchID, objPath string, metadata map[string]string, contentType string, contents io.ReadSeeker) (*api.ObjectStats, error) {
```
Comment (itaiad200) [inline pkg/api/helpers/upload.go:20]:
```suggestion
func ClientUploadDirect(ctx context.Context, client api.ClientWithResponsesInterface, repoID, branchID, objPath string, metadata map[string]string, contentType string, contents io.ReadSeeker) (*api.ObjectStats, error) {
```

**Review by itaiad200 — APPROVED** [review]
Thanks, code looks great! If it's possible, please add lakectl tests.
My only comment regards to product requirements - do we really need all 3 options for using lakectl?

**Comment by nopcoder** [comment]
@itaiad200 I missed you comments - I'll answer here and open a new pr if needed

File: pkg/api/helpers/upload.go (hunk 2 above)
Comment (nopcoder) [reply pkg/api/helpers/upload.go:20]: Addressed at https://github.com/treeverse/lakeFS/pull/5112

File: api/swagger.yml (hunk 1 above)
Comment (nopcoder) [reply api/swagger.yml:863]: PreSign is enabled by default, meaning if you request for get physical address you will also get pre-sign url.
There is no block adapter configuration - but not all adapters supports this option, like: mem and local.
Added this code to identify it while testing. At runtime you will get an error.

**Pull request merged**